io/plugins/eu.esdihumboldt.hale.io.jdbc.ui/src/eu/esdihumboldt/hale/io/jdbc/ui/SchemasRetrievalPage.java

```diff
@@ -106,71 +118,146 @@ public boolean updateConfiguration(ImportProvider provider) {
-		layoutData.widthHint = SWT.DEFAULT;
-		layoutData.heightHint = 8 * schemaList.getList().getItemHeight();
-		schemaList.getControl().setLayoutData(layoutData);
-		setPageComplete(false);
+		this.page = page;
 	}
 
 	/**
 	 * @see eu.esdihumboldt.hale.ui.HaleWizardPage#onShowPage(boolean)
 	 */
 	@Override
 	protected void onShowPage(boolean firstShow) {
-		if (firstShow) {
-			setPageComplete(true);
-		}
 		try {
+			// load configuration
+			loadConfiguration();
+
+			// if page is enable for driver
+			if (!isEnable) {
+				disposeControl();
+				// message
+				Label label = new Label(page, SWT.WRAP);
+				label.setText("Schema retrieval is not enabled for selected driver.");
+				label.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, false, 2, 1));
+				page.layout(true, true);
+				setMessage("");
+				setPageComplete(true);
+				return;
+			}
+
+			if (firstShow) {
+				// when first time loaded
+				createComponent();
+			}
+			else {
+				// When user come to page again
+				disposeControl();
+				schemas.clear();
+				createComponent();
+			}
+
+			// Get schemas using selected database driver
 			getSchemas();
-			schemaList.setInput(schemas);
-		} catch (SQLException e) {
+			schemaTable.setInput(schemas);
 
-			e.printStackTrace();
+			if (multipleSelection) {
+				// if multiple Selection is enabled for driver then by default
+				// select all.
+				schemaTable.setAllChecked(true);
+				setPageComplete(true);
+				setMessage("Please select one or multiple schemas.");
+			}
+			else {
+				setMessage("Please select one schema.");
+			}
+		} catch (SQLException e) {
+			log.error(e.getMessage(), e);
 		}
 
 	}
 
-	/**
-	 * 
-	 * @throws SQLException this
-	 * 
-	 */
+	private void createComponent() {
+		if (multipleSelection)
+			schemaTable = CheckboxTableViewer.newCheckList(page,
+					SWT.MULTI | SWT.V_SCROLL | SWT.H_SCROLL | SWT.BORDER);
+		else
+			schemaTable = CheckboxTableViewer.newCheckList(page,
+					SWT.SINGLE | SWT.V_SCROLL | SWT.H_SCROLL | SWT.BORDER | SWT.BORDER);
+
+		schemaTable.setLabelProvider(new LabelProvider() {
+
+			@Override
+			public String getText(Object element) {
+				return element.toString();
+			}
+
+		});
+		schemaTable.setContentProvider(ArrayContentProvider.getInstance());
+		GridData layoutData = new GridData(SWT.FILL, SWT.FILL, true, false, 2, 1);
+		layoutData.widthHint = SWT.DEFAULT;
+		layoutData.heightHint = 8 * schemaTable.getTable().getItemHeight();
+		schemaTable.getControl().setLayoutData(layoutData);
+
+		schemaTable.addSelectionChangedListener(new ISelectionChangedListener() {
+
+			@Override
+			public void selectionChanged(SelectionChangedEvent event) {
+				setPageComplete(validate());
+			}
+		});
+
+		schemaTable.addFilter(new ViewerFilter() {
+
+			@Override
+			public boolean select(Viewer viewer, Object parentElement, Object element) {
+				String value = (String) element;
+
+				if (value.equalsIgnoreCase("information_schema") || value.startsWith("pg_"))
```

Comment:
This is logic specific to a certain type of database. The UI should not have to care about that. The schema selector for a database should only return what is appropriate.